app/models/user.rb

```diff
@@ -0,0 +1,15 @@
+class User < ApplicationRecord
+  before_save :downcase_email
+
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :name, presence: true
+  validates :email, presence: true, uniqueness: true, format: {with: VALID_EMAIL_REGEX}
+
+  has_secure_password
+
+  private
+
+  def downcase_email
+    self.email.downcase
```

Comment:
```suggestion
    email.downcase!
```
so sánh giúp anh .downcase với .downcase! nhé

Reply:
downcase! sẽ trả về nil nếu đầu vào ko có thay đổi. ví dụ "abc".downcase! sẽ trả về nil